internal/ast/astutil/walk.go

```diff
@@ -0,0 +1,257 @@
+package astutil
+
+import "github.com/influxdata/flux/ast"
+
+// A Visitor's Visit method is invoked for each node encountered by Walk.
+// If the result visitor w is not nil, Walk visits each of the children
+// of node with the visitor w, followed by a call of w.Done(n).
+// This will not visit nil nodes.
+type Visitor interface {
+	Visit(ast.Node) Visitor
+	Done(ast.Node)
+}
+
+func Walk(v Visitor, n ast.Node) {
+	switch n := n.(type) {
+	case *ast.ArrayExpression:
+		if n == nil {
+			return
+		}
+		if w := v.Visit(n); w != nil {
+			for _, element := range n.Elements {
+				w.Visit(element)
```

Comment:
Note, these calls to w.Visit do not recurse, so the Walk implementation as it stands now only visits direct children, but not any grand children.